code/game/objects/items/weapons/implants/implant_explosive.dm

```diff
@@ -31,13 +31,45 @@
 	to_chat(imp_in, "<span class='danger'>You activate your microbomb bio-chip.</span>")
 //If the delay is short, just blow up already jeez
 	if(delay <= 7)
-		explosion(src, heavy, medium, weak, weak, flame_range = weak)
-		if(imp_in)
-			imp_in.gib()
-		qdel(src)
+		self_destruct()
 		return
 	timed_explosion()
 
+// Micro/macro-bomb users (typically nukies) should have all of their
+// (destructible) equipment destroyed on activation.
+/obj/item/implant/explosive/proc/self_destruct()
+	if(!imp_in)
+		return
+
+	explosion(src, heavy,medium,weak,weak, flame_range = weak)
+
+	// Iterate over the implantee's contents and take out indestructible
+	// things to avoid having to worry about containers and recursion
+	var/preserved_items = list()
+	var/destructed_items = list()
+	// In case something happens to the implantee between now and the
+	// self-destruct
+	var/current_location = get_turf(imp_in)
+
+	for(var/obj/item/I in imp_in.get_contents())
+		if(I)
+			if(I == src) // Don't delete ourselves prematurely
+				continue
+			if(I.resistance_flags & INDESTRUCTIBLE)
+				preserved_items += I
+			else
+				destructed_items += I
```

Comment:
is there any reason you don't perform the forcemoves and qdels here?

Reply:
We could forcemove the indestructible items in this loop definitely, but would it be safe to do the qdels here? If there's an indestructible item in the qdel'd item and that indestructible item hasn't yet shown up while iterating, is there a reference to it anywhere? Does its presence in the result of the `get_contents` call count?

Reply:
That is a good point I did not think of. The current system you use now is good, you could consider the forcemove in this loop just to avoid iterating over things redundantly. 

I know you go over this a little bit in the documentation, but could you leave a small comment just explaining that you want to first remove all indestructible items before starting qdel's for this reason?

Reply:
Done. Moved around and cleaned up the docs to be more hopefully useful.